beforematch-security-and-privacy-assessment.md

```diff
@@ -0,0 +1,167 @@
+# beforematch event security and privacy self-review questionnaire
+
+The beforematch event ([explainer](https://github.com/WICG/display-locking/blob/master/explainer-beforematch.md)) is fired on elements before scrolling occurs containing matching text in these three cases:
+* When the fragment is changed to scroll to an element by id, including on navigation to a url which includes a fragment.
+  * I will refer to this case as "ElementFragment."
+* When the [ScrollToTextFragment](https://github.com/WICG/ScrollToTextFragment) feature finds text and scrolls to it.
+  * I will refer to this case as "ScrollToTextFragment."
+* When the find-in-page (ctrl+f) feature is used to search for text.
+  * I will refer to this case as "find-in-page."
+
+## Questions to Consider
+
+### 2.1. What information might this feature expose to Web sites or other parties, and for what purposes is that exposure necessary?
+
+This feature may expose hints as to what the user is searching for to webpages. I wrote more detailed considerations about this in the "Privacy Considerations" section at the bottom of this document.
+
+This exposure is necessary in order to fulfill the use case of displaying content that was searched for, as described in [the beforematch explainer](https://github.com/WICG/display-locking/blob/master/explainer-beforematch.md).
+
+### 2.2. Is this specification exposing the minimum amount of information necessary to power the feature?
+
+For the use case of modifying the render-subtree css property on matching elements, less information would be exposed if we just automatically changed the property to expand the search result when searching instead. Although you could use the render-subtree property and a mutation observer to see where matches occur to find out which elements matched what the user searched for, this is more restrictive than firing the beforematch event every time a search occurs regardless of the render-subtree property.
```

Comment:
s/render-subtree/subtree-visibility/

So, I think you're answering the question kind of holistically here considering the whole subtree-visibility / display locking project. That is, the argument is that we could expose less information if we didn't have the `beforematch` event at all, which is true. However, when answering the question "are we exposing the minimum amount of information _to make `beforematch` possible_", I think the answer would be different. WDYT?

To put this differently, I think we can discuss whether `beforematch` is the right event to expose separately from discussing the privacy and security implications of `beforematch`; and this document is meant to do the latter.

Reply:
I see, you have a good point. Without thinking about subtree-visibility, I'm not really sure what there is to talk about here. Is there another use case I could or should think from? Should I just change the wording?

Reply:
I think this should just say something like

"""
Yes, this is exposing only the information required for the `beforematch` event to function as specified.
"""

In other words, It doesn't leak anything extra information, it just exposes the information that it is meant to expose by design.

Reply:
Sounds good